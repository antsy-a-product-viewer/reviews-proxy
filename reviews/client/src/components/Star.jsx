import React from 'react';
import _ from 'underscore';
import styles from './css/reviewStyles.css.js';

const Star = (props) => {
  return (
    <div style={{
      alignSelf: 'center'
    }}>
      {_.times(props.stars, (n) =>{
        return (
          <img className={props.starClass} key={n} style={styles.star} src="https://s3-us-west-1.amazonaws.com/anstyicons/icon-star-512.png"></img>
        );
      })}
    </div>
  );
};

export default Star;
