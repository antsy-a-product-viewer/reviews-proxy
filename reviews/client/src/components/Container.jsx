import React from 'react';
import axios from 'axios';
import _ from 'underscore';
import Star from './Star.jsx';
import ReviewPhotosContainer from './ReviewPhotosContainer.jsx';
import Modal from './Modal.jsx';
import containerStyles from './css/containerStyles.css.js';
import ReviewContainer from './ReviewContainer.jsx';
import Button from './Button.jsx';

class Container extends React.Component {
  constructor(props) {
    super(props);
    this.openModal = this.openModal.bind(this);
    this.closeModal = this.closeModal.bind(this);
    this.getData = this.getData.bind(this);
    this.getReviews = this.getReviews.bind(this);
    this.getReviewImages = this.getReviewImages.bind(this);
    this.getAverageStars = this.getAverageStars.bind(this);
    this.showMore = this.showMore.bind(this);
    this.visitItem = this.visitItem.bind(this);
    this.reviewImageClick = this.reviewImageClick.bind(this);
    this.state = {
      storeId: 0,
      averageStars: 0,
      reviews: [],
      reviewImages: [],
      showModal: false,
      show: 4,
      modalReview: null
    };
  }

  componentDidMount() {
    this.getData();
  }

  openModal(reviewImgIndex) {
    this.setState({
      modalReview: reviewImgIndex,
      showModal: !this.state.showModal
    });
  }

  closeModal(e) {
    if (e.target.id === 'modal-overlay' || e.target.id === 'close-modal') {
      this.setState({
        showModal: !this.state.showModal,
        modalReview: null
      });
    }
  }

  getData() {
    var endpoint = window.location.pathname + 'store_id';
    axios.get(endpoint)
      .then((res) => {
        var storeId = res.data[0].store_id;
        this.setState({
          storeId: storeId
        });
        this.getReviews(storeId);
        this.getReviewImages(storeId);
      })
      .catch((err) => {
        console.log(err);
      });
  }

  getReviews(storeId) {
    axios.get(`/stores/${storeId}/reviews`)
      .then((res) => {
        this.getAverageStars(res.data);
        this.setState({
          reviews: res.data
        });
      })
      .catch((err) => {
        console.log(err);
      }); 
  }

  getReviewImages(storeId) {
    axios.get(`/stores/${storeId}/review_images`)
      .then((res) => {
        this.setState({
          reviewImages: res.data
        });
      })
      .catch((err) => {
        console.log(err);
      });
  }

  getAverageStars(reviews) {
    var total = 0;
    for (var i = 0; i < reviews.length; i++) {
      total += reviews[i].stars;
    }
    var average = Math.floor(total / reviews.length);
    this.setState({
      averageStars: average
    });
  }

  showMore() {
    this.setState({
      show: 20
    });
  }
  
  visitItem(productId) {
    window.location.pathname = `/product/${productId}`;
  }

  reviewImageClick(id) {
    this.setState({
      modalReview: _.findIndex(this.state.reviewImages, {review_id: parseInt(id)}),
      showModal: true
    });
  }
  
  render() {
    return (
      <div style={containerStyles.container}>
        <div style={{display: 'flex', marginBottom: '-10px'}}>
          <div style={containerStyles.header}>
            Reviews
          </div>
          <Star starClass="averageStars" stars={this.state.averageStars}/>
          <div style={containerStyles.centeredDiv}>
            ({this.state.reviews.length})
          </div>
        </div>
        <ReviewContainer reviews={this.state.reviews} limit={this.state.show} showPrice="false" visitItem={this.visitItem} imageClick={this.reviewImageClick}/>
        <Button currentNumber={this.state.show} showMore={this.showMore} totalReviews={this.state.reviews.length}/>
        <ReviewPhotosContainer reviewImages={this.state.reviewImages} openModal={this.openModal}/>
        <Modal showModal={this.state.showModal} onClose={this.closeModal} visitItem={this.visitItem} review={this.state.reviewImages[this.state.modalReview]}/>
      </div>
    );
  }
}


export default Container;