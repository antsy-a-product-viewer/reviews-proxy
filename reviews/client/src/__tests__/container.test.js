import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { mount, shallow } from 'enzyme';
import Container from '../components/Container.jsx';
import Star from '../components/Star.jsx';

configure({ adapter: new Adapter() });

it('should render correctly with no props', () => {
  const container = shallow(<Container />);
  
  expect(container).toMatchSnapshot();
});

it('should call getData during componentDidMount', () => {
  const getData = jest.spyOn(Container.prototype, 'getData');
  const container = mount(<Container />);

  expect(getData).toHaveBeenCalledTimes(1);
});

it('should set the ReviewContainer limit to 20 when showMore is clicked', () => {
  const showMore = jest.spyOn(Container.prototype, 'showMore');
  const container = shallow(<Container />);
  container.instance().showMore();
  container.update();

  expect(container.find('ReviewContainer').prop('limit')).toEqual(20);
});

const reviews = [
  {
      "review_id": 532,
      "user_img": "https://s3-us-west-1.amazonaws.com/antsyuserimages/00010.jpg",
      "user_name": "Jalen Paucek",
      "created_at": "2014-02-15T21:43:09.000Z",
      "stars": 5,
      "review": "You can't connect the pixel without overriding the auxiliary RSS card!",
      "item_id": 72,
      "item_img": "https://s3-us-west-1.amazonaws.com/antsyitemsimages/00072.jpg",
      "item_name": "Ergonomic Concrete Tuna"
  },
  {
      "review_id": 119,
      "user_img": "https://s3-us-west-1.amazonaws.com/antsyuserimages/00027.jpg",
      "user_name": "Zella Blick",
      "created_at": "2014-03-14T22:37:45.000Z",
      "stars": 4,
      "review": "Try to hack the SQL transmitter, maybe it will navigate the multi-byte bus!",
      "item_id": 30,
      "item_img": "https://s3-us-west-1.amazonaws.com/antsyitemsimages/00030.jpg",
      "item_name": "Handmade Concrete Salad"
  },
  {
      "review_id": 35,
      "user_img": "https://s3-us-west-1.amazonaws.com/antsyuserimages/00024.jpg",
      "user_name": "Deangelo Schuppe",
      "created_at": "2014-03-27T15:37:49.000Z",
      "stars": 0,
      "review": "I'll generate the cross-platform COM firewall, that should driver the XSS pixel!",
      "item_id": 91,
      "item_img": "https://s3-us-west-1.amazonaws.com/antsyitemsimages/00091.jpg",
      "item_name": "Rustic Metal Chair"
  },
  {
      "review_id": 398,
      "user_img": "https://s3-us-west-1.amazonaws.com/antsyuserimages/00014.jpg",
      "user_name": "Stacey Spencer",
      "created_at": "2014-05-01T06:50:00.000Z",
      "stars": 3,
      "review": "We need to reboot the primary XML circuit!",
      "item_id": 4,
      "item_img": "https://s3-us-west-1.amazonaws.com/antsyitemsimages/00004.jpg",
      "item_name": "Generic Steel Soap"
  },
  {
      "review_id": 167,
      "user_img": "https://s3-us-west-1.amazonaws.com/antsyuserimages/00022.jpg",
      "user_name": "Avis Quitzon",
      "created_at": "2015-07-21T23:47:15.000Z",
      "stars": 3,
      "review": "If we hack the system, we can get to the AGP bandwidth through the primary HTTP firewall!",
      "item_id": 95,
      "item_img": "https://s3-us-west-1.amazonaws.com/antsyitemsimages/00095.jpg",
      "item_name": "Unbranded Metal Fish"
  }
]
