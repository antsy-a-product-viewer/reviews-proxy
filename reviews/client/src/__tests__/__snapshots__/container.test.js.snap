// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly with no props 1`] = `
<div
  style={
    Object {
      "fontFamily": "Gill Sans",
      "width": 600,
    }
  }
>
  <div
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <div
      style={
        Object {
          "alignSelf": "center",
          "fontSize": 20,
          "marginRight": 10,
        }
      }
    >
      Reviews
    </div>
    <Star
      starClass="averageStars"
      stars={0}
    />
    <div
      style={
        Object {
          "alignSelf": "center",
          "marginLeft": 10,
        }
      }
    >
      (
      0
      )
    </div>
  </div>
  <ReviewContainer
    limit={4}
    reviews={Array []}
    showPrice="false"
    visitItem={[Function]}
  />
  <Button
    currentNumber={4}
    showMore={[Function]}
    totalReviews={0}
  />
  <ReviewPhotosContainer
    openModal={[Function]}
    reviewImages={Array []}
  />
  <Modal
    onClose={[Function]}
    showModal={false}
  />
</div>
`;
